### PR TITLE
Mes 4843 AND Mes 4842 late and early test start

### PR DESCRIPTION
**Pull request by turnerc1**
## Description
Checks to see if popup dialog comes if starting a late test and early test and to see if it can continue onto the declaration page.
## Checklist

- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Comment by turonekt** [comment]
There is no functional difference between test that starts early on time or late. 
Maybe scenario could finish with only a check if 'late test' modal appears and declaration page (test was started) appears.

**Comment by turnerc1** [comment]
> There is no functional difference between test that starts early on time or late.
> Maybe scenario could finish with only a check if 'late test' modal appears and declaration page (test was started) appears.

@turonekt 
Should I just make a single story then that instead of going through the whole journey instead have it look at if the popup message comes up as well as if it can continue to the next page?

File: test/e2e/features/cat-b/early_and_late_popup.feature [hunk 1]
```diff
@@ -0,0 +1,15 @@
+@catb
+Feature: early and late popup
```
Comment (jono-basford-bjss) [inline test/e2e/features/cat-b/early_and_late_popup.feature]: more descriptive Feature name

**Comment by turnerc1** [comment]
> There is no functional difference between test that starts early on time or late.
> Maybe scenario could finish with only a check if 'late test' modal appears and declaration page (test was started) appears.

@turonekt I have done what you have said and have only made it so that it checks to see if it appears and checks to see if it goes onto the declaration page. 
I have done the same as MES-4842 which is to check for an early start, as the stories are together

File: test/e2e/features/cat-b/96-early_and_late_popup.feature [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@catb
+Feature: early and late popup
```
Comment (jono-basford-bjss) [inline test/e2e/features/cat-b/96-early_and_late_popup.feature]: more descriptive feature name

File: test/e2e/features/cat-b/96-early_and_late_popup.feature [hunk 3]
```diff
@@ -0,0 +1,13 @@
+@catb
```
Comment (jono-basford-bjss) [inline test/e2e/features/cat-b/96-early_and_late_popup.feature]: add @regression tag ( see other feature files)

File: test/e2e/step-definitions/journal-steps.ts [hunk 4]
```diff
@@ -150,3 +176,25 @@ const closeCandidateDetailsDialog = () => {
...
+
+const startingEarlyTest = (candidateName) => {
+  const buttonElement = getElement(by.xpath(`//button/span/h3[text()[normalize-space(.) = "Start test"]]
+    [ancestor::ion-row/ion-col/ion-grid/ion-row/ion-col/candidate-link/div/button/span/
+    h3[text() = "${candidateName}"]]`));
+  clickElement(buttonElement);
+
+  const testEarlyDialog = getElement(by.xpath(`/html/body/ion-app/ion-modal/div/
+  journal-early-start-modal/ion-card/modal-alert-title/ion-row[2]/ion-col/h2`));
+  return expect(testEarlyDialog.isPresent()).to.eventually.be.true;
```
Comment (jono-basford-bjss) [inline test/e2e/step-definitions/journal-steps.ts]: can you create a helper function for the expect assertion

File: test/e2e/step-definitions/journal-steps.ts [hunk 5]
```diff
@@ -150,3 +176,25 @@ const closeCandidateDetailsDialog = () => {
...
+
+const startingExpiredTest = (candidateName) => {
+  const buttonElement = getElement(by.xpath(`//button/span/h3[text()[normalize-space(.) = "Start test"]]
+    [ancestor::ion-row/ion-col/ion-grid/ion-row/ion-col/candidate-link/div/button/span/
+    h3[text() = "${candidateName}"]]`));
+  clickElement(buttonElement);
+
+  const testExpireDialog = getElement(by.xpath(`/html/body/ion-app/ion-modal/div/journal-rekey-modal/ion-card/
+    modal-alert-title/ion-row[2]/ion-col/h2`));
+  return expect(testExpireDialog.isPresent()).to.eventually.be.true;
```
Comment (jono-basford-bjss) [inline test/e2e/step-definitions/journal-steps.ts:199]: can you create a helper function for the expect assertion

File: test/e2e/step-definitions/journal-steps.ts [hunk 6]
```diff
@@ -108,31 +134,31 @@ Then('I should have a category {string} test for {string}', (category, candidate
...
 
-  const seatValue = getElement(
-    by.xpath(`//vehicle-details/div/span/span[text() = 'Seats: ']/following-sibling::span
+    const seatValue = getElement(
+      by.xpath(`//vehicle-details/div/span/span[text() = 'Seats: ']/following-sibling::span
     [ancestor::ion-grid/ion-row/ion-col/ion-grid/ion-row/ion-col/candidate-link/div/button/span/h3
       [text() = "${candidateName}"]]`));
-  return expect(seatValue.getText()).to.eventually.equal(seats);
-});
+    return expect(seatValue.getText()).to.eventually.equal(seats);
```
Comment (jono-basford-bjss) [inline test/e2e/step-definitions/journal-steps.ts:160]: can you create a helper function for the expect assertion

File: test/e2e/step-definitions/journal-steps.ts [hunk 7]
```diff
@@ -108,31 +134,31 @@ Then('I should have a category {string} test for {string}', (category, candidate
...
-(candidateName, length, width, height, seats) => {
-  const lengthValue = getElement(
-    by.xpath(`//vehicle-details/div/span/span[text()= 'L: ']/following-sibling::span
+  (candidateName, length, width, height, seats) => {
+    const lengthValue = getElement(
+      by.xpath(`//vehicle-details/div/span/span[text()= 'L: ']/following-sibling::span
     [ancestor::ion-grid/ion-row/ion-col/ion-grid/ion-row/ion-col/candidate-link/div/button/span/h3
       [text() = "${candidateName}"]]`));
-  expect(lengthValue.getText()).to.eventually.equal(length);
+    expect(lengthValue.getText()).to.eventually.equal(length);
```
Comment (jono-basford-bjss) [inline test/e2e/step-definitions/journal-steps.ts:142]: white space?

File: test/e2e/step-definitions/journal-steps.ts [hunk 8]
```diff
@@ -108,31 +134,31 @@ Then('I should have a category {string} test for {string}', (category, candidate
 });
 
 Then('The vehicle for {string} has length {string}, width {string}, height {string} and seats {string}',
-(candidateName, length, width, height, seats) => {
-  const lengthValue = getElement(
-    by.xpath(`//vehicle-details/div/span/span[text()= 'L: ']/following-sibling::span
+  (candidateName, length, width, height, seats) => {
+    const lengthValue = getElement(
```
Comment (jono-basford-bjss) [inline test/e2e/step-definitions/journal-steps.ts:138]: white space?

**Review by jono-basford-bjss — CHANGES_REQUESTED** [review]

File: test/e2e/features/cat-b/early_and_late_popup.feature (hunk 1 above)
Comment (turnerc1) [reply test/e2e/features/cat-b/early_and_late_popup.feature]: Done

File: test/e2e/features/cat-b/96-early_and_late_popup.feature (hunk 2 above)
Comment (turnerc1) [reply test/e2e/features/cat-b/96-early_and_late_popup.feature]: Done

File: test/e2e/features/cat-b/96-early_and_late_popup.feature (hunk 3 above)
Comment (turnerc1) [reply test/e2e/features/cat-b/96-early_and_late_popup.feature]: Done

File: test/e2e/step-definitions/journal-steps.ts (hunk 7 above)
Comment (turnerc1) [reply test/e2e/step-definitions/journal-steps.ts:142]: Done.

File: test/e2e/step-definitions/journal-steps.ts (hunk 8 above)
Comment (turnerc1) [reply test/e2e/step-definitions/journal-steps.ts:138]: Done.

File: test/e2e/step-definitions/journal-steps.ts (hunk 4 above)
Comment (turnerc1) [reply test/e2e/step-definitions/journal-steps.ts]: Done.

File: test/e2e/step-definitions/journal-steps.ts (hunk 5 above)
Comment (turnerc1) [reply test/e2e/step-definitions/journal-steps.ts:199]: Done.

File: test/e2e/step-definitions/journal-steps.ts (hunk 6 above)
Comment (turnerc1) [reply test/e2e/step-definitions/journal-steps.ts:160]: Done.

**Review by jono-basford-bjss — APPROVED** [review]

**Pull request closed without merging**